bpf/lib/nodeport.h

```diff
@@ -1998,9 +1998,46 @@ static __always_inline int encap_geneve_dsr_opt4(struct __ctx_buff *ctx, int l3_
 }
 #endif /* DSR_ENCAP_MODE */
 
+/* remove_dsr_ip_opt_v4 removes the DSR IP Option from the packet and 
+ * recalculates the IPv4 checksum.
+ *
+ * Must run from a context where the DSR option presence has been identified.
+ */
+static __always_inline int
+remove_dsr_ip_opt_v4(struct __ctx_buff *ctx, struct iphdr *ip4) {
+	struct dsr_opt_v4 opt;
+	__u32 iph_old, iph_new, sum_l3;
+
+	if (ctx_load_bytes(ctx, ETH_HLEN + sizeof(struct iphdr), &opt, sizeof(opt)) < 0)
+		return DROP_INVALID;
+	if (opt.type != DSR_IPV4_OPT_TYPE)
+		return CTX_ACT_OK;
```

Comment:
You could just have `nodeport_extract_dsr_v4()` pass in the options instead, and remove these parts.

Reply:
Done.